pyscriptjs/src/main.ts

```diff
@@ -248,65 +249,112 @@ export class PyScriptApp {
-                    logger.error(`Cannot find plugin on Python module ${modulename}! Python plugins \
-modules must contain a "plugin" attribute. For more information check the plugins documentation.`);
-                }
-            } catch (e) {
-                //Should we still export full error contents to console?
-                handleFetchError(<Error>e, singleFile);
+            if (singleFile.endsWith('.py')) {
+                await this.fetchPythonPlugin(runtime, singleFile);
+            } else if (singleFile.endsWith('.js')) {
+                await this.fetchJSPlugin(singleFile);
+            } else {
+                throw new UserError(
+                    ErrorCode.BAD_PLUGIN_FILE_EXTENSION,
+                    `Unable to load plugin from '${singleFile}'. ` +
+                    `Plugins need to contain a file extension and be ` +
+                    `either a python or javascript file.`);
             }
+            logger.info("All plugins fetched");
+        }
+    }
+
+    /**
+     * Fetch a javascript plugin from a filePath, it gets a blob from the
+     * fetch and creates a file from it, then we create a URL from the file
+     * so we can import it as a module.
+     *
+     * This allow us to instantiate the imported plugin with the default
+     * export in the module (the plugin class) and add it to the plugins
+     * list with `new importedPlugin()`.
+     *
+     * @param filePath - path to the js file to fetch
+     */
+    async fetchJSPlugin(filePath: string) {
+        const pluginBlob = await (await robustFetch(filePath)).blob();
+        const blobFile = new File([pluginBlob], "plugin.js", {type: "text/javascript"});
+        const fileUrl = URL.createObjectURL(blobFile);
+
+        const module = await import(fileUrl);
+        // Note: We have to put module.default in a variable
+        // because we have seen weird behaviour when doing
+        // new module.default() directly.
+        const importedPlugin = module.default;
+
+        // If the imported plugin doesn't have a default export
+        // it will be undefined, so we throw a user error, so
+        // an alter banner will be created.
+        if (importedPlugin === undefined) {
+            throw new UserError(
+                ErrorCode.NO_DEFAULT_EXPORT,
+                `Unable to load plugin from '${filePath}'. ` +
+                `Plugins need to contain a default export.`);
+        } else {
+            this.plugins.add(new importedPlugin);
+        }
+    }
+
+    /**
+     * Fetch python plugins from a filePath, saves it on the FS and import
+     * it as a module, executing any plugin define the module scope.
+     *
+     * @param runtime - runtime that will execute the plugins
+     * @param filePath - path to the python file to fetch
+     */
+    async fetchPythonPlugin(runtime: Runtime, filePath: string) {
+        const plugins = this.config.plugins;
+        logger.info('Python plugins to fetch: ', plugins);
```

Comment:
Don't think we actually use this anymore, true? Since fetchPythonPlugin is called once for each `.js` or `.py` file?